### PR TITLE
[sb] update conn str

### PR DESCRIPTION
**Pull request by l0lawrence**
porting changes from https://github.com/Azure/azure-sdk-for-python/pull/34103 and subsequent pr https://github.com/Azure/azure-sdk-for-python/pull/35623/files to servicebus

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
My Python is rusty, but the general form and the slugs look good to me.

File: sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py [hunk 1]
```diff
@@ -248,14 +248,10 @@ def from_connection_string(
...
-            shared_access_key_name,
-            shared_access_key,
-            _,
-            token,
-            token_expiry,
-        ) = _parse_conn_str(conn_str)
+        (endpoint, shared_access_key_name, shared_access_key, _, token, token_expiry, emulator) = _parse_conn_str(
+            conn_str
+        )
+        kwargs["use_tls"] = not emulator
```
[the remark this answers is not in the document]
Comment (l0lawrence) [reply sdk/servicebus/azure-servicebus/azure/servicebus/aio/management/_management_client_async.py:254]: https://learn.microsoft.com/en-us/azure/service-bus-messaging/overview-emulator#known-limitations oops will remove :)

**Comment by l0lawrence** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**